### ISSUE TITLE
no timeout: Solr request failed with 404; Operation not permitted

### ISSUE THREAD

**Issue by szabgab** [markov2/perl5-Apache-Solr#3]
As I was trying to run the tests it got stuck in an **infinite** loop printing:

```
alert: Solr request failed with 404; Operation not permitted
 at lib/Apache/Solr.pm line 935
```

I guess my setup is incorrect and it cannot access `solr`, but I think this should have some timeout, shouldn't it?

**Comment by markov2** [comment]
When SOLR_TEST_SERVER is empty or undefined, those tests should be skipped.  But one was "skip_all" was missing in the last release.

I have cleaned-up the error for the upcoming release (1.09, today) as well.
Do you use Apache::Solr yourself?

**Comment by markov2** [comment]
Released

**Comment by szabgab** [comment]
The context for the issue is that I was trying to set up GitHub Actions to run the tests on every push. (See my [Daily CI challenge](https://dev.to/szabgab/the-2022-december-ci-challenge-5dof).
In order to make it really valuable I wanted to set up a solr service, but apparently either I did it incorrectly or I set SOLR_TEST_SERVER incorrectly. That's how I got into this infinite loop.

I don't really know how this module should work, but it looked strange that it got into an infinite loop. It did not look correct even if in my case it was my fault.

I'll investigate the topic further.
